docs/DeveloperGuide.md

```diff
@@ -1041,5 +1049,86 @@ user experience of WedLog.
 The implementation of this enhancement will be done by first designing a different colour scheme with higher contrast for WedLog, and then changing the colour scheme of the UI components
 by changing the relevant style attributes in the `.css` files.
 
+--------------------------------------------------------------------------------------------------------------------
+
+## **Appendix D: Effort**
+
+This section documents the effort required to evolve AB3 into WedLog.
+
+**Replacing `Person` with `Guest` and `Vendor` classes**
 <br>
 
+This involved: 
+* Creating `Guest` and `Vendor` classes.
+* Rewriting all existing code dealing with `Person` to handle `Guest` and `Vendor`.
+  * E.g. `EditCommand` into `GuestEditCommand` and `VendorEditCommand`.
+* Splitting parsing of vendor and guest commands to their respective Parser classes `GuestCommandParser` and `VendorCommandParser`.
+* Updating UI to display both lists.
+* Removing all depreciated classes handling the `Person` class.
+
+This was time and effort intensive as:
+* Tracking two entities as opposed to one in AB3 increased the complexity of our project.
+* From this point onwards, we had to create two of each command and command parsers to handle `Guest` and `Vendor` separately.
+
+**Altering most fields to become Optional**
+
+Our app allows fields like `Phone`, `Email`, `Address` and more to be empty, which AB3 did not.
+
+This increased complexity of our application as features like `add`, `edit` and `filter` had to account for more 
+variations in values.
+
+
+**Enhancing `Guest` class with new parameters**
+
+We enhanced the `Guest` class to track additional information not covered in the original `Person` class. This involved:
+* Introducing the `TableNumber`, `RsvpStatus` and `DietaryRequirements` classes and integrating them into existing 
+features like `add` and `edit`.
+
+This change was challenging as it required lots of in-depth design discussions on how to best represent the information.
+  * E.g. For `RsvpStatus` class: We debated on the appropriate amount of flexibility to give users, and eventually 
+  settled on restricting acceptable values for `RsvpStatus` to `Yes`, `No`, and `Unknown`.
+  * E.g. For `DietaryRequirements` class: We initially stored the information as a string, but later adapted it into a 
+  tag system to facilitate UI design and filtering.
```

Comment:
```suggestion
  * For `RsvpStatus` class: We debated on the appropriate amount of flexibility to give users, and eventually 
  settled on restricting acceptable values for `RsvpStatus` to `Yes`, `No`, and `Unknown`.
  * For `DietaryRequirements` class: We initially stored the information as a string, but later adapted it into a 
  tag system to facilitate UI design and filtering.
```

not sure if we should just leave out the E.g. since it is just 2 bullet points?

Reply:
Good point!